From 17a5cf8f70b86d8a2195562e2d9dcccb4fa35c83 Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Tue, 31 Oct 2017 20:05:58 -0700
Subject: [PATCH] fix configure option parsing

Don't over-write with_openssl etc by trying to set to $enablevar

Upstream-Status: Pending

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 configure.ac | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 82e6d2d..823e6c1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -56,17 +56,17 @@ fi
 dnl Options for overriding TLS checks
 AC_ARG_WITH([openssl],
   AS_HELP_STRING([--without-openssl],[disable checking for openssl]),
-  [with_openssl=$enableval],
-  [with_openssl=yes]
+  [],
+  [with_openssl=auto]
 )
 AC_ARG_WITH([gnutls],
   AS_HELP_STRING([--without-gnutls],[disable checking for GNU TLS]),
-  [with_gnutls=$enableval],
-  [with_gnutls=yes]
+  [],
+  [with_gnutls=auto]
 )
 
 dnl Check OpenSSL
-if test "x$with_openssl" = "xyes"; then
+if test "x$with_openssl" != "xno"; then
   PKG_CHECK_MODULES([OPENSSL], openssl >= 0.9.8, have_openssl=yes, have_openssl=no)
   if test "x$have_openssl" = "xyes"; then
     LIBOPENSSL_CFLAGS="$OPENSSL_CFLAGS"
@@ -78,7 +78,7 @@ if test "x$with_openssl" = "xyes"; then
 fi
 
 dnl Check GNU TLS
-if test "x$with_gnutls" = "xyes"; then
+if test "x$with_gnutls" != "xno"; then
   if test "x$have_openssl" != "xyes"; then
     PKG_CHECK_MODULES([GNUTLS], gnutls >= 2.0.0, have_gnutls=yes, have_gnutls=no)
     if test "x$have_gnutls" = "xyes"; then
@@ -94,7 +94,7 @@ fi
 dnl Option for overriding Python check
 AC_ARG_ENABLE([python],
   AS_HELP_STRING([--disable-python],[disable checking for Python bindings]),
-  [with_python=$enableval],
+  [],
   [with_python=yes]
 )
 
-- 
1.9.1

